src/OcisResource.php

```diff
@@ -73,6 +117,104 @@ private function getMetadata(ResourceMetadata $property): array|string
         return $metadata[$property->getKey()];
     }
 
+    /**
+     * gets all possible permissions for the resource
+     * @return array<SharingRole>
+     * @throws BadRequestException
+     * @throws ForbiddenException
+     * @throws HttpException
+     * @throws NotFoundException
+     * @throws UnauthorizedException
+     * @throws InvalidResponseException
+     */
+    public function getRoles(): array
+    {
+        $guzzle = new Client(
+            Ocis::createGuzzleConfig($this->connectionConfig, $this->accessToken)
+        );
+
+        if (array_key_exists('drivesPermissionsApi', $this->connectionConfig)) {
+            $apiInstance = $this->connectionConfig['drivesPermissionsApi'];
+        } else {
+            $apiInstance = new DrivesPermissionsApi(
+                $guzzle,
+                $this->graphApiConfig
+            );
```

Comment:
in other places we do this kind of dependency injection for unit tests through special setters, e.g. https://github.com/owncloud/ocis-php-sdk/blob/main/src/Ocis.php#L173-L179
If we are happy to use `connectionConfig` for that, we should refactor the other injections accordingly (in a different PR)